### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by patrickkrootjes**
This pull request introduces several major improvements to the high school activity management system, including new activities, enhanced participant management (both backend and frontend), and comprehensive automated tests. The most important changes are grouped below by theme.

**Backend enhancements and new features:**

* Added several new activities (`Soccer Team`, `Swimming Club`, `Art Studio`, `Drama Club`, `Debate Team`, `Science Olympiad`) to the `activities` data structure in `src/app.py`.
* Implemented a new API endpoint (`DELETE /activities/{activity_name}/unregister`) to allow students to unregister from activities, with appropriate validation and error handling.
* Improved signup logic to prevent duplicate registrations by checking if a student is already signed up before adding them.

**Frontend improvements for participant management:**

* Updated the activities list UI to display participants for each activity, including a "remove participant" (delete) button for each student, and a message when there are no participants.
* Implemented frontend logic to handle participant removal: clicking the delete button triggers a confirmation and, if confirmed, calls the new backend unregister endpoint and refreshes the UI.
* Ensured the activity selection dropdown is reset before repopulating, preventing duplicate options.
* After signup or unregister actions, the frontend now refreshes the activities list to reflect participant changes immediately.

**Styling and user experience:**

* Added and refined CSS styles for the participants section, participant list, delete button, and "no participants" message to improve UI clarity and usability.

**Testing and reliability:**

* Added a comprehensive test suite (`tests/test_app.py`) covering all major API endpoints, business logic, and edge cases, including signup, unregister, and activity constraints, with fixtures to reset state between tests.

### CODE REVIEW

File: src/static/styles.css [hunk 1]
```diff
@@ -142,3 +142,67 @@ footer {
...
+  color: #555;
+  font-size: 0.9em;
+}
+
+.no-participants {
+  margin: 8px 0 0 0;
+  color: #999;
+  font-style: italic;
+  font-size: 0.9em;
+}
```
[the remark this answers is not in the document]
Comment (patrickkrootjes) [reply src/static/styles.css:202]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/patrickkrootjes/redesigned-waffle/pull/2#discussion_r2659025841)

[the remark this answers is not in the document]
Comment (patrickkrootjes) [reply src/static/styles.css:202]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/patrickkrootjes/redesigned-waffle/pull/2#discussion_r2659025841)
